### PR TITLE
Use webpack to bundle the extension

### PR DESCRIPTION
**Pull request by giordano**
For reference, I followed [VS Code documentation about webpack](https://code.visualstudio.com/api/working-with-extensions/bundling-extension#using-webpack)

### CODE REVIEW

File: .vscodeignore [hunk 1]
```diff
@@ -1,4 +1,11 @@
 .vscode/**
 .vscode-test/**
 .gitignore
-vsc-extension-quickstart.md
+python/**
+node_modules/**
```
Comment (ageorgou) [inline .vscodeignore:5]: Is it okay to ignore even the (runtime) dependencies? Does webpack include them in the `dist` version?

File: .vscodeignore [hunk 2]
```diff
@@ -1,4 +1,11 @@
...
+out/**
+src/**
+*.vsix
+azure-pipelines.yml
+tsconfig.json
+webpack.config.js
```
Comment (ageorgou) [inline .vscodeignore:11]: Should we also add `.git/`?

File: package.json [hunk 3]
```diff
@@ -101,6 +101,10 @@
     },
     "scripts": {
         "clean": "rm -rf ./out",
+        "vscode:prepublish": "webpack --mode production",
+        "webpack": "webpack --mode development",
+        "webpack-dev": "webpack --mode development --watch",
+        "test-compile": "tsc -p ./",
```
Comment (ageorgou) [inline package.json:107]: Is this something invoked by webpack? I see it's in the guide but is that only to showcase testing and the launch configuration?

File: .vscodeignore (hunk 1 above)
Comment (ageorgou) [reply .vscodeignore:5]: Never mind, I read more :)

**Review by ageorgou — COMMENTED** [review]
Seems sensible to me, to the extent that I understand it!

File: .vscodeignore (hunk 2 above)
Comment (giordano) [reply .vscodeignore:11]: That's probably ignored by default:
```console
% unzip -l nisaba-0.0.1.vsix
Archive:  nisaba-0.0.1.vsix
  Length      Date    Time    Name
---------  ---------- -----   ----
     2046  2021-08-16 17:23   extension.vsixmanifest
      527  2021-08-16 17:23   [Content_Types].xml
      231  2021-08-16 17:23   extension/CHANGELOG.md
   252401  2021-08-16 17:23   extension/dist/extension.js
       55  2021-08-16 17:23   extension/dist/extension.js.LICENSE.txt
  1008056  2021-08-16 17:23   extension/dist/extension.js.map
      823  2020-06-16 14:57   extension/language-configuration.json
      784  2020-06-16 14:53   extension/media/preview_dark.svg
      784  2020-06-16 14:53   extension/media/preview_light.svg
      768  2021-07-16 11:47   extension/media/validation_dark.svg
      762  2021-07-16 11:47   extension/media/validation_light.svg
     4320  2021-08-16 17:22   extension/package.json
     1301  2021-08-16 17:23   extension/README.md
     3332  2021-03-03 12:06   extension/syntaxes/atf.tmLanguage.json
---------                     -------
  1276190                     14 files
```

File: package.json (hunk 3 above)
Comment (giordano) [reply package.json:107]: I'm not sure if/where this command would be called.  It's also very similar to the `compile`

**Review by ageorgou — APPROVED** [review]

**Pull request merged**